### PR TITLE
feat(disclaimer): SDK - Support audio disclaimer for recording

### PR DESCRIPTION
**Pull request by BBigLee**
# COMPLETES #< INSERT LINK TO ISSUE >
[SDK - Support audio disclaimer for recording](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-420195)

## This pull request addresses
1. Add SERVER_AUDIO_ANNOUNCEMENT_SUPPORTED toggle into deviceCapabilities for /loci/call, and /loci/{lid}/participant calls
2. Add unit tests for the changes
3. Update sample page

## by making the following changes

< DESCRIBE YOUR CHANGES >

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation update
- [ ] Tooling change
- [x] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [x] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/@webex/internal-plugin-locus/src/locus.js [hunk 1]
```diff
@@ -265,12 +268,20 @@ const Locus = WebexPlugin.extend({
       throw new Error('options.correlationId is required');
     }
 
+    // Support server side audio disclaimer
+    const deviceCapabilities = options.deviceCapabilities || [];
+    if (!deviceCapabilities.includes(SERVER_AUDIO_ANNOUNCEMENT_SUPPORTED)) {
```
Comment (mickelr) [inline packages/@webex/internal-plugin-locus/src/locus.js:273]: Need a feature toggle to control the deviceCapabilities

**Review by mickelr — CHANGES_REQUESTED** [review]

File: packages/@webex/plugin-meetings/src/meeting/request.ts [hunk 2]
```diff
@@ -156,17 +167,25 @@ export default class MeetingRequest extends StatelessWebexPlugin {
...
-      deviceCapabilities = [...deviceCapabilities, ANNOTATION.ANNOTATION_ON_SHARE_SUPPORTED];
+      deviceCapabilities.push(ANNOTATION.ANNOTATION_ON_SHARE_SUPPORTED);
     }
-    if (deviceCapabilities.length > 0) {
-      body.deviceCapabilities = deviceCapabilities;
+
+    // Support server side audio disclaimer
+    if (isJoining) {
+      body.locale = locale || DEFAULT_LOCALE;
+      if (!deviceCapabilities.includes(SERVER_AUDIO_ANNOUNCEMENT_SUPPORTED)) {
```
Comment (mickelr) [inline packages/@webex/plugin-meetings/src/meeting/request.ts:180]: Need a feature toggle to control the deviceCapabilities

File: packages/@webex/internal-plugin-locus/src/locus.js [hunk 3]
```diff
@@ -397,11 +408,19 @@ const Locus = WebexPlugin.extend({
       throw new Error('locus.correlationId or options.correlationId is required');
     }
 
+    // Support server side audio disclaimer
+    const deviceCapabilities = options.deviceCapabilities || [];
```
Comment (mickelr) [inline packages/@webex/internal-plugin-locus/src/locus.js:412]: When or in which case the code will be executed?

**Review by mickelr — CHANGES_REQUESTED** [review]

File: packages/@webex/plugin-meetings/src/meeting/request.ts [hunk 4]
```diff
@@ -127,6 +137,10 @@ export default class MeetingRequest extends StatelessWebexPlugin {
       roapMessage,
       preferTranscoding,
       breakoutsSupported,
+      homerDisclaimerEnabled,
+      isJoining,
+      locale,
+      deviceCapabilities = [],
```
Comment (mickelr) [inline packages/@webex/plugin-meetings/src/meeting/request.ts:143]: There is already a deviceCapabilities definition in the function

File: packages/@webex/plugin-meetings/src/meeting/util.ts [hunk 5]
```diff
@@ -107,6 +107,10 @@ MeetingUtil.joinMeeting = (meeting, options) => {
       preferTranscoding: !meeting.isMultistream,
       asResourceOccupant: options.asResourceOccupant,
       breakoutsSupported: options.breakoutsSupported,
+      locale: options.locale,
+      deviceCapabilities: options.deviceCapabilities,
```
Comment (mickelr) [inline packages/@webex/plugin-meetings/src/meeting/util.ts:111]: Don't need add the deviceCapabilities parameter

**Review by mickelr — CHANGES_REQUESTED** [review]

File: packages/@webex/plugin-meetings/src/meeting/request.ts (hunk 4 above)
Comment (BBigLee) [reply packages/@webex/plugin-meetings/src/meeting/request.ts:143]: It's better to allow user to pass deviceCapabilities into SDK, so we don't need to update SDK every time when we want to and a new toggle for deviceCapabilities props.

File: packages/@webex/plugin-meetings/src/meeting/util.ts (hunk 5 above)
Comment (BBigLee) [reply packages/@webex/plugin-meetings/src/meeting/util.ts:111]: Same as above

File: packages/@webex/plugin-meetings/src/meeting/request.ts [hunk 6]
```diff
@@ -156,17 +170,25 @@ export default class MeetingRequest extends StatelessWebexPlugin {
...
-      deviceCapabilities = [...deviceCapabilities, ANNOTATION.ANNOTATION_ON_SHARE_SUPPORTED];
+      deviceCapabilities.push(ANNOTATION.ANNOTATION_ON_SHARE_SUPPORTED);
     }
-    if (deviceCapabilities.length > 0) {
-      body.deviceCapabilities = deviceCapabilities;
+
+    // Support server side audio disclaimer
+    if (homerDisclaimerEnabled && isJoining) {
+      body.locale = locale || DEFAULT_LOCALE;
+      if (!deviceCapabilities.includes(SERVER_AUDIO_ANNOUNCEMENT_SUPPORTED)) {
```
Comment (mickelr) [inline packages/@webex/plugin-meetings/src/meeting/request.ts:183]: I think deviceCapabilities should not be related with isJoining status

Comment (BBigLee) [reply packages/@webex/plugin-meetings/src/meeting/request.ts:183]: Only joining a meeting action need the SERVER_AUDIO_ANNOUNCEMENT_SUPPORTED toggle, not all situations. But this function will be called under many conditions. So it's necessary to add a status to tell them apart. @mickelr

Comment (BBigLee) [reply packages/@webex/plugin-meetings/src/meeting/request.ts:183]: And only for the value SERVER_AUDIO_ANNOUNCEMENT_SUPPORTED that the deviceCapabilities prop is related, not for other cases. It doesn't affect other capabilities logics.

**Review by mickelr — APPROVED** [review]

**Pull request merged**